libsql-replication/src/injector/mod.rs

```diff
@@ -161,7 +171,24 @@ impl Injector {
     }
 
     fn begin_txn(&mut self) -> Result<(), Error> {
-        let conn = self.connection.lock();
+        let mut conn = self.connection.lock();
```

Comment:
Could you add a description to the PR to explain why we need to replace the connection?